src/Http/Routing/src/Patterns/RoutePatternParser.cs

```diff
@@ -568,7 +568,7 @@ private string DebuggerToString()
                 }
                 else
                 {
-                    return _template.Substring(0, _index) + "|" + _template.Substring(_index);
+                    return string.Concat(_template.Substring(0, _index), "|", _template.Substring(_index));
```

Comment:
I would think it wants you to do 
```suggestion
                    return string.Concat(_template.AsSpan(0, _index), "|", _template.AsSpan(_index));
```